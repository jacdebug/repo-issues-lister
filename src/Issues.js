import React from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import { compose, renderComponent, lifecycle, branch } from "recompose";
import { selectIssues } from "./reducers";
import { loadIssues } from "./actions";
import Loading from "./Loading";

export const Issues = ({ issues }) => (
  <ul>{issues.map(({ id, title }) => <li key={id}>{title}</li>)}</ul>
);

Issues.propTypes = {
  issues: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      title: PropTypes.string.isRequired
    })
  ).isRequired
};

export default compose(
  connect(
    (state, props) => ({
      issues: selectIssues(state, props.repo)
    }),
    dispatch => ({
      loadIssues: repo => dispatch(loadIssues(repo))
    })
  ),
  lifecycle({
    componentDidMount() {
      if (!this.props.issues) {
        this.props.loadIssues(this.props.repo);
      }
    }
  }),
  branch(props => !props.issues, renderComponent(() => <Loading />))
)(Issues);
