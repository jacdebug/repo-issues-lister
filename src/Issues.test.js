import React from 'react';
import { shallow } from 'enzyme';
import { Issues } from './Issues';

describe('<Issues />', () => {
  it('should render', () => {
    const component = shallow(
      <Issues
        issues={[
          {
            id: 1,
            title: 'issue title test first',
          },
          {
            id: 2,
            title: 'issue title test second',
          },
        ]}
      />
    );

    expect(component).toMatchSnapshot();
    expect(component.find('ul').length).toBe(1);
    expect(component.find('li').length).toBe(2);
  });
});
