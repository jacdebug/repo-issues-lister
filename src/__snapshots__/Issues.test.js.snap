// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Issues /> should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    issues={
        Array [
            Object {
              "id": 1,
              "title": "issue title test first",
            },
            Object {
              "id": 2,
              "title": "issue title test second",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li>
          issue title test first
</li>,
        <li>
          issue title test second
</li>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": "issue title test first",
        },
        "ref": null,
        "rendered": "issue title test first",
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "host",
        "props": Object {
          "children": "issue title test second",
        },
        "ref": null,
        "rendered": "issue title test second",
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li>
            issue title test first
</li>,
          <li>
            issue title test second
</li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": "issue title test first",
          },
          "ref": null,
          "rendered": "issue title test first",
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "host",
          "props": Object {
            "children": "issue title test second",
          },
          "ref": null,
          "rendered": "issue title test second",
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
